### PR TITLE
feat(db): drop retry logic

### PR DESCRIPTION
**Pull request by alexirion10**
if `restoring` DB is good to drop
if `catch(SmoException e) when(e.Message.Contains("Cannot access properties or methods"))` DB is good to drop

### CODE REVIEW

File: src/SmoFacade/Database.cs [hunk 1]
```diff
@@ -134,5 +117,31 @@ public void MultiUserMode()
       _database.DatabaseOptions.UserAccess = DatabaseUserAccess.Multiple;
       _database.Alter(TerminationClause.RollbackTransactionsImmediately);
     }
+
+    private void ThrowIfUnreadyToDrop()
+    {
+      if(Restoring)
+        return; // restoring state means we're good to drop
```
Comment (AdityaNPL) [inline src/SmoFacade/Database.cs:124]: If it's restoring, shouldn't we wait until it is finished?

File: src/SmoFacade/Database.cs [hunk 2]
```diff
@@ -134,5 +117,31 @@ public void MultiUserMode()
...
+            throw
+              new Exception($"Cannot kill the database {Name} until it has been removed from the AvailabilityGroup");
+          if(_database.AvailabilityDatabaseSynchronizationState >
+             AvailabilityDatabaseSynchronizationState.NotSynchronizing)
+            throw new
+              Exception($"Cannot kill the database {Name} until AvailabilityDatabaseSynchronizationState is inaccessible");
+        }
+        catch(
+          PropertyCannotBeRetrievedException) { } // good here, AvailabilityDatabaseSynchronizationState unavailable means it has no state
+        catch(SmoException e) when(e.Message.Contains("Cannot access properties or methods")) { }
```
Comment (AdityaNPL) [inline src/SmoFacade/Database.cs:143]: Should we log the error message in these cases?
I just feel catching and ignoring exceptions may lead to headaches down the line when trying to debug prod issues

**Review by AdityaNPL — COMMENTED** [review]
Have a few questions

File: src/SmoFacade/Database.cs (hunk 1 above)
Comment (alexirion10) [reply src/SmoFacade/Database.cs:124]: no - restoring state doesn't necessarily mean that it's currently writing a backup.  
Restoring state happens on secondary nodes when the DB is removed from the availability group. 

See:  https://docs.microsoft.com/en-us/sql/database-engine/availability-groups/windows/remove-a-primary-database-from-an-availability-group-sql-server?view=sql-server-ver15#FollowUp

File: src/SmoFacade/Database.cs (hunk 2 above)
Comment (alexirion10) [reply src/SmoFacade/Database.cs:143]: I see no need to log here, these are cases where we can safely proceed

**Review by AdityaNPL — APPROVED** [review]
LGTM!

**Comment by alexirion10** [comment]
@Kaneraz - could you take a look at this?

**Review by Kaneraz — COMMENTED** [review]
Oops, late to hit submit here. The comments we discussed.

**Review by Kaneraz — APPROVED** [review]
Thanks, looks good

**Pull request merged**